server/src/main/java/io/spine/server/ServerEnvironment.java

```diff
@@ -313,11 +326,17 @@ public void close() throws Exception {
         if (tracerFactory != null) {
             tracerFactory.close();
         }
-        if (transportFactory != null) {
-            transportFactory.close();
+        if (transportFactories.production()
```

Comment:
To elimitate the repeating `if (... .production().isPresent()) { ...}` please introduce the corresponding API to the `EnvironmentDependantValue` itself. 

I can see at least four cases only in this class.